site/content/contribute/server/schema-migration-guide.md

```diff
@@ -21,8 +32,13 @@ The problem arises when in some databases, for some tables, due to various techn
 
 ### I need to make a schema change. What do I do?
 
-1. Every store is initialized in a `newXXXStore` function in `store/sqlstore/XXX_store.go` file. It should have a `db.AddTableWithName` call. Add your column to that table with the right `ColMap` param. Add other constraints as necessary.
-2. Modify the `store/sqlstore/upgrade.go` file. Scroll to the bottom, and add your migration there. (This method is going to change in the near future).
+1. Add appropriate sql script file with the statements you want to run into the migrations directory. Which is located in `{project_dir}/db/migrations/{driver_name}/`. Do not forget to add script for `mysql` and `postgres`.
+2. To embed the script run `make migrations-bindata`.
+3. You can run the mattermost-server binary and the tool will automatically apply the migration if it is required. And the migration name will be saved into `db_migrations` table.
+
+### My migration has failed. What do I do?
+1. If you think your migration is applied, and you want to revert changes: You can run the down script to rollback in a clean way. You can use morph CLI to apply down migrations.
+2. The migration has been shipped in a release and you want to apply fixes: Instead of changing the existing script, you should add a new one so that `db_migrations` will stay consistent. You can edit the existing migration to be a no-op for future releases in this case.
```

Comment:
```suggestion
2. If the migration has been shipped in a release and you want to apply fixes, instead of changing the existing script, you should add a new one so that `db_migrations` will stay consistent. You can edit the existing migration to be a no-op for future releases in this case.
```